prdoc/pr_6255.prdoc

```diff
@@ -0,0 +1,26 @@
+title: '[pallet-child-bounties] Index child bounties by parent bounty'
+doc:
+- audience: Runtime Dev
+  description: |
+    Index child bounties by their parent bounty, ensuring that their indexes are independent of
```

Comment:
IMP the description isn't super clear. Say a opensquare team member looking at this, how do they suppose to know what exactly they need to do without actually read all the code in this PR?

Reply:
updated